core/crypto/eddsa/blinded25519_test.go

```diff
@@ -0,0 +1,133 @@
+package eddsa
+
+import (
+	"filippo.io/edwards25519"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"io"
+	"math/rand"
+	"testing"
+	"testing/quick"
+)
+
+var identity_element = [32]byte{1, 0}
+
+func check_public_key(pk *PublicKey) bool {
+	// here we do scalar multiplication with L as the scalar;
+	// the result should be 1 if the public key is valid.
+	var result [32]byte
+	order_64 := [64]byte{0xed, 0xd3, 0xf5, 0x5c, 0x1a, 0x63, 0x12, 0x58, 0xd6, 0x9c,
+		0xf7, 0xa2, 0xde, 0xf9, 0xde, 0x14, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x10}
+	order_sc, _ := new(edwards25519.Scalar).SetUniformBytes(order_64[:])
+	pkA_n, _ := new(edwards25519.Point).SetBytes(pk.Bytes())
+	pkA_n.ScalarMult(order_sc, pkA_n)
+	copy(result[:], pkA_n.Bytes())
```

Comment:
why is this copy necessary for the comparison?

Reply:
in general we don't use == for byte comparisons because it only works on arrays AND in this particular case pkA_n.Bytes() returns a byte slice. Therefore it would be more appropriate to just compare the byte slices directly with bytes.Equal (from the go std libarry "bytes" module)... and in a production module (this one is a test), we'd use hmac.Equal or the "subtle" module for constant time comparison.

Reply:
so to fix it... 

1. remove the var result declaration
2. remove the copy
3. perform comparison with bytes.Equal and return the bool

Reply:
I think I was getting confused by slice/byte stuff in Golang when I wrote this, I've updated with @david415's suggestion